src/main/java/calculator/model/delimiter/CustomDelimiterProcessor.java

```diff
@@ -0,0 +1,53 @@
+package calculator.model.delimiter;
+
+import static calculator.config.CustomDelimiterPattern.*;
+
+import java.util.Optional;
+
+import calculator.validation.InputValidator;
+
+public class CustomDelimiterProcessor {
+	public String removeCustomDelimiterPattern(String input) {
+		if (hasCustomDelimiterPattern(input)) {
+			int endIndex = findEndIndex(input);
+
+			return input.substring(endIndex + END.getPatternLength());
+		}
+
+		return input;
+	}
+
+	public Optional<String> extractCustomDelimiter(String input) {
+		if (!hasCustomDelimiterPattern(input)) {
+			return Optional.empty();
+		}
+		int startIndex = findStartIndex(input);
+		int endIndex = findEndIndex(input);
+
+		return Optional.of(extractCustomDelimiter(input, startIndex, endIndex));
+	}
+
+	private boolean hasCustomDelimiterPattern(String input) {
+		return input.contains(START.getPattern()) && input.contains(END.getPattern());
+	}
```

Comment:
```suggestion
	private boolean containsCustomDelimiter(String input) {
		return input.contains(START.getPattern()) && input.contains(END.getPattern());
	}
```

has보다는 자바에서 관습적으로 쓰이는 contains를 써보시는 게 어떨까요?
추가로 pattern이라는 말이 오히려 혼란스러울 수도 있을 것 같아요.
`//` `\\n`이 있다면 커스텀 구분자가 있다. 라고 받아들여도 무방하지 않을까요?

Reply:
`contains`라는 키워드가 좀 더 직관적인 것 같긴 하네요! 문자열을 처리하는 만큼 "포함하고 있다"라는 말이 "가지고 있다"보다 좀 더 의미가 잘 통하는 것 같습니다.

`pattern` 같은 경우는 사실 쓰면서도 고민을 했던 것 같아요! 분명 사용하는 단어긴 한데, 리뷰어들이 이 단어만 보고 바로 `//`, `\n`에 대해 떠올릴 수 있을까?에 대한 의문이 있었던 것 같아요.

> // \\n이 있다면 커스텀 구분자가 있다. 라고 받아들여도 무방하지 않을까요?

`hasCustomDelimiterPattern()`은 단순히 코드의 내용을 요약하려다보니 생긴 결과물인데, 생각해보니 "커스텀 구분자를 포함하고 있다"로 치환시키는 게 확실히 더 명료한 것 같습니다! 네이밍에 대해서도 @takoyakimchi 님께 많이 배워가는 것 같습니다 😀